cinema/permissions.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        return bool(
+            (
+                request.method in SAFE_METHODS
+                and request.user
+                and request.user.is_authenticated
+            )
+            or (
+                request.user and request.user.is_staff
+            )
+
+        )
+
+
+class PermissionMixin:
```

Comment:
it'd be better to move this mixin into another module(mixins.py or smth like that), and not to mix up mixins and permission